MapboxNavigation/NavigationMapView.swift

```diff
@@ -225,6 +225,11 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         addGestureRecognizer(mapTapGesture)
     }
     
+    open override func layoutMarginsDidChange() {
+        super.layoutMarginsDidChange()
+        enableFrameByFrameCourseViewTracking(for: 3)
```

Comment:
Perhaps there’s a race condition in which this method gets called before [the map view updates its center coordinate](https://github.com/mapbox/mapbox-gl-native/blob/c5a21ca8f282f4ded76663ea46d79c368e384205/platform/ios/src/MGLMapView.mm#L1091) to reflect the margin change, so that by the time we convert the user location to a center point within the (non)animation block, the center point still reflects the old margins.

https://github.com/mapbox/mapbox-navigation-ios/blob/a0e5dbc3ed63754a4311964d5c41ba4636d17988/MapboxNavigation/NavigationMapView.swift#L332-L334

Temporarily enabling frame-by-frame course tracking ensures that the view moves as soon as the map adjusts to the new content insets (since that requires the map view to render another frame):

https://github.com/mapbox/mapbox-navigation-ios/blob/a0e5dbc3ed63754a4311964d5c41ba4636d17988/MapboxNavigation/NavigationMapView.swift#L261-L264

Temporarily reenabling frame-by-frame course tracking isn’t the end of the world, though if it really comes down to this race condition, then I’d think a shorter duration would still be sufficient to work around it.